src/core_plugins/kibana/public/dashboard/index.js

```diff
@@ -29,34 +28,75 @@ const app = uiModules.get('app/dashboard', [
 ]);
 
 uiRoutes
-.defaults(/dashboard/, {
-  requireDefaultIndex: true
-})
-.when('/dashboard', {
-  template: indexTemplate,
-  resolve: {
-    dash: function (savedDashboards, config) {
-      return savedDashboards.get();
+  .defaults(/dashboard/, {
+    requireDefaultIndex: true
+  })
+  .when('/dashboard', {
+    template: function ($location) {
+      if ($location.new === true) {
+        return indexTemplate;
+      }
+
+      return false;
+    },
+    resolve: {
+      dash: function (savedDashboards, config, kbnUrl, Notifier, $location, $route) {
+        let defaultDashboard = config.get('dashboard:defaultDashboard', '');
+
+        let notify = new Notifier({
+          location: 'Dashboard'
+        });
+
+        if ($location.search().new === true) {
+          return savedDashboards.get();
+        }
+
+        function forceNew() {
+          $location.search('new', true);
+          $route.reload();
+        }
+
+        if (defaultDashboard !== '' && typeof defaultDashboard !== 'undefined') {
```

Comment:
Why not just `if (defaultDashboard)`?  I suspect we wouldn't want to try to get a `null`, `NaN` or `false` dashboard or anything of that sort.